### PR TITLE
19 menulinkwithicon

### PR DESCRIPTION
**Pull request by 0xMazout**

### CODE REVIEW

File: src/components/LinkIcon/LinkIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React, { ComponentPropsWithoutRef, useState } from "react";
+
+export interface LinkIconProps extends ComponentPropsWithoutRef<"button"> {
+  children?: React.ReactNode;
+  icon?: string;
+}
+
+export function LinkIcon(props: LinkIconProps) {
+  const [isActive, setisActive] = useState(false);
+  let styleDiv = `flex mas-menu-default  ${isActive ? " bg-primary" : ""}
```
Comment (pivilartisant) [inline src/components/LinkIcon/LinkIcon.tsx]: Maybe you can add a pointer cursor on hover ?

File: src/components/LinkIcon/LinkIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export function LinkIcon(props: LinkIconProps) {
+  const [isActive, setisActive] = useState(false);
+  let styleDiv = `flex mas-menu-default  ${isActive ? " bg-primary" : ""} 
+                  rounded-lg p-3 items-center text-neutral hover:bg-tertiary 
+`;
+  const onclickHandler = () => {
+    props.onClick;
+    setisActive((current) => !current);
```
Comment (pivilartisant) [inline src/components/LinkIcon/LinkIcon.tsx]: Are you trying to set a style for when the div is pressed ?

File: src/components/LinkIcon/LinkIcon.test.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  test("it should render", () => {
+    render(<LinkIcon>something</LinkIcon>);
+
+    let element = screen.getByTestId("link-icon");
+
+    expect(element).toBeTruthy();
+    expect(element.textContent).toBe("something");
+  });
+
+  test("it should have mas-menu-default class", () => {
```
Comment (mazmassa) [inline src/components/LinkIcon/LinkIcon.test.tsx:15]: We should add tests for click link action...

File: src/components/Buttons/LinkIcon/MenuLink.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import React, { ComponentPropsWithoutRef, useState } from "react";
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.tsx:1]:
```suggestion
import { ReactNode , ComponentPropsWithoutRef, useState } from "react";
```

File: src/components/Buttons/LinkIcon/MenuLink.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+export interface MenuLinkProps extends ComponentPropsWithoutRef<"button"> {
+  children?: React.ReactNode;
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.tsx:4]:
```suggestion
  children?: ReactNode;
```

File: src/components/Buttons/LinkIcon/MenuLink.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  icon?: JSX.Element;
+  color?: string;
+  onClick?: () => void; // Add onClick prop of type function that doesn't return anything
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.tsx:7]: Not sure we need that... if we use ...props we let that for the parent component. Isn't it!? Check how we did in button component.

**Review by mazmassa — APPROVED** [review]

File: src/components/Buttons/LinkIcon/MenuLink.test.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+    let element = screen.getByTestId("link-icon");
+
+    expect(element).toBeInTheDocument();
+    expect(element.textContent).toBe("something");
+  });
+
+  test("it should have a click link action", () => {
+    const somethingClicked = jest.fn();
+    render(<MenuLink onClick={somethingClicked}>something</MenuLink>); // Pass onClick prop
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.test.tsx:17]:
```suggestion

    render(<MenuLink onClick={somethingClicked}>something</MenuLink>);
    
```

File: src/components/Buttons/LinkIcon/MenuLink.test.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  test("it should render", () => {
+    render(<MenuLink>something</MenuLink>);
+
+    let element = screen.getByTestId("link-icon");
+
+    expect(element).toBeInTheDocument();
+    expect(element.textContent).toBe("something");
+  });
+
+  test("it should have a click link action", () => {
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.test.tsx:15]:
```suggestion
  test("it should fires an onClic action", () => {
```

File: src/components/Buttons/LinkIcon/MenuLink.test.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    const somethingClicked = jest.fn();
+    render(<MenuLink onClick={somethingClicked}>something</MenuLink>); // Pass onClick prop
+    let element = screen.getByTestId("link-icon");
+    fireEvent.click(element);
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.test.tsx:19]:
```suggestion

    fireEvent.click(element);
```

File: src/components/Buttons/LinkIcon/MenuLink.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  function onClickHandler(
+    e: MouseEvent<HTMLDivElement, globalThis.MouseEvent>
+  ): void {
+    setIsActive(!isActive);
+    onClick?.(e);
+  }
+
+  return (
+    <div
+      data-testid="link-icon"
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.tsx:32]:
```suggestion
      data-testid="menu-link"
```

File: src/components/Buttons/LinkIcon/MenuLink.test.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    let element = screen.getByTestId("link-icon");
+
+    expect(element).toBeInTheDocument();
+    expect(element.textContent).toBe("something");
+  });
+
+  test("it should have a click link action", () => {
+    const somethingClicked = jest.fn();
+    render(<MenuLink onClick={somethingClicked}>something</MenuLink>); // Pass onClick prop
+    let element = screen.getByTestId("link-icon");
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.test.tsx:18]:
```suggestion
    let element = screen.getByTestId("menu-link");
```

File: src/components/Buttons/LinkIcon/MenuLink.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import { MenuLink } from "./MenuLink";
+import { BiHomeAlt } from "react-icons/bi";
+
+export default { title: "buttons", component: MenuLink };
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.stories.tsx:4]:
```suggestion
export default { title: "Components/MenuLink", component: MenuLink };
```

File: src/components/Buttons/LinkIcon/MenuLink.tsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+export interface MenuLinkProps extends ComponentPropsWithoutRef<"div"> {
+  children?: ReactNode;
+  icon?: JSX.Element;
+  color?: string;
+}
+
+export function MenuLink(props: MenuLinkProps) {
+  const [isActive, setIsActive] = useState(false);
+  const { onClick } = props;
```
Comment (mazmassa) [inline src/components/Buttons/LinkIcon/MenuLink.tsx:16]:
```suggestion
    const { onClick } = props;
    const [isActive, setIsActive] = useState(false);
```

**Review by pivilartisant — APPROVED** [review]

**Comment by SlnPons** [comment]
@0xMazout there is a problem with the colours. It doesn't match the design system. See below: 
Dark theme: 
* default : OK
image
* hover: KO
image
It should be: 
image

* pressed: KO 
At pressed, the text colour changes. It should be the colour button. 

Light theme: 
Same problems with inverted colours. 

Can you please correct these on all the menu link please?

**Pull request merged**